pandas/_libs/tslib.pyx

```diff
@@ -551,7 +551,7 @@ cpdef array_to_datetime(
                             raise ValueError('Cannot mix tz-aware with '
                                              'tz-naive values')
                         if isinstance(val, _Timestamp):
-                            iresult[i] = val.as_unit("ns").value
+                            iresult[i] = (<_Timestamp>val)._as_creso(NPY_FR_ns).value
```

Comment:
Does `.value` naturally not return a Timestamp here?

Reply:
`.value` returns an `int64_t`

Reply:
Ah gotcha. The cast isn't needed though right? Even `pydatetime_to_dt64` returns an int64_t so probably better to remove this

Reply:
ill just revert this, as the code currently there should work fine (just less performant)